legacy/builder/wipeout_build_path_if_build_options_changed.go

```diff
@@ -40,9 +40,13 @@ func (s *WipeoutBuildPathIfBuildOptionsChanged) Run(ctx *types.Context) error {
 	previousBuildOptionsJson := ctx.BuildOptionsJsonPrevious
 
 	var opts *properties.Map
-	var prevOpts *properties.Map
 	json.Unmarshal([]byte(buildOptionsJson), &opts)
```

Comment:
For coherency, this one should be tested too. Would the message be similar ?
`build.options.json nvalid, rebuilding all`

Reply:
The situation is different with `ctx.BuildOptionsJson` compared to `ctx.BuildOptionsJsonPrevious` because `ctx.BuildOptionsJsonPrevious` contains data that comes from an external source but `ctx.BuildOptionsJson` is generated from the internal data. From what I can see, this could only contain bad data if something had gone wrong inside Arduino CLI or the `properties` package.

So, even though recovering on bad data in `ctx.BuildOptionsJsonPrevious` is definitely the right thing to do, it seems to me that it's most appropriate to fail on bad data in `ctx.BuildOptionsJson`.